src/org/zaproxy/admin/HelpGenerator.java

```diff
@@ -0,0 +1,168 @@
+/*
+ * Zed Attack Proxy (ZAP) and its related class files.
+ * 
+ * ZAP is an HTTP/HTTPS proxy for assessing web application security.
+ * 
+ * Licensed under the Apache License, Version 2.0 (the "License"); 
+ * you may not use this file except in compliance with the License. 
+ * You may obtain a copy of the License at 
+ * 
+ *   http://www.apache.org/licenses/LICENSE-2.0 
+ *   
+ * Unless required by applicable law or agreed to in writing, software 
+ * distributed under the License is distributed on an "AS IS" BASIS, 
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. 
+ * See the License for the specific language governing permissions and 
+ * limitations under the License. 
+ */
+package org.zaproxy.admin;
+
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Map.Entry;
+
+import org.vafer.jdeb.shaded.compress.io.FileUtils;
+
+public class HelpGenerator {
+	
+	private static final File TEMPLATE_DIR = new File("src/org/zaproxy/admin/resources/help"); 
+	
+	private static void recursiveCopy(File srcFile, File destFile, Map<String, String> tokens) {
+		if (srcFile.isDirectory()) {
+			if (destFile.mkdirs()) {
+				System.out.println("Created " + destFile.getAbsolutePath());
+			} else {
+				System.out.println("Failed to create " + destFile.getAbsolutePath());
+				return;

```

Comment:
Sorry, this should throw an exception instead. This method is recursive. failing to create the directories would lead to further problems.
One thing that we need to do is check if `destFile` already exist before attempting to create it/them (`mkdirs` creates intermediate directories too, which could fail when calling `recursiveCopy` again).